app/models/db/MetricsFilters.scala

```diff
@@ -1,59 +1,68 @@
 package models.db
 
+import com.github.tototoshi.slick.PostgresJodaSupport._
+import models.db.Schema.DBMetric
 import org.joda.time.DateTime
 import org.joda.time.format.ISODateTimeFormat
+import slick.jdbc.PostgresProfile.api._
+import slick.lifted.{LiteralColumn, Rep}
 
 import scala.util.control.NonFatal
 
-case class DateRange (from: Option[DateTime], to: Option[DateTime])
+case class DateRange (from: DateTime, to: DateTime)
 
 case class MetricsFilters(
-                           dateRange: Option[DateRange],
-                           desk: Option[String],
-                           startingSystem: Option[String] = None,
-                           filtersList: List[String]
+  dateRange: Option[DateRange] = None,
+  desk: Option[String] = None,
+  originatingSystem: Option[OriginatingSystem] = None
 )
 
 object MetricsFilters {
+  private val TrueOptCol : Rep[Option[Boolean]] = LiteralColumn(Some(true))
+
   def apply(queryString: Map[String, Seq[String]]): MetricsFilters =
     MetricsFilters(
       dateRange = extractDateRange(queryString),
       desk = getOptionFromQS("desk", queryString),
-      filtersList = List("dateRange", "desk", "startingSystem")
+      originatingSystem = OriginatingSystem.withNameOption(getOptionFromQS("originatingSystem", queryString).getOrElse(""))
     )
 
   private def getOptionFromQS(key: String, qs: Map[String, Seq[String]]): Option[String] = qs.get(key).flatMap(_.headOption)
 
-  private def parseDate(dateString: String): Option[DateTime] = dateString match {
-    case "" => None
-    case _ =>
-      try { Some(ISODateTimeFormat.dateTimeParser.parseDateTime(dateString)) }
-      catch { case NonFatal(_) => None }
-  }
-
   private def extractDateRange(qs: Map[String, Seq[String]]): Option[DateRange] = {
     val date: Option[String] = getOptionFromQS("date", qs)
     val dateRange: Option[DateRange] = date.fold(None: Option[DateRange])(getDateRangeFromDateString)
 
     dateRange match {
       case None =>
-        val start = getOptionFromQS("startDate", qs).flatMap(parseDate)
-        val end = getOptionFromQS("endDate", qs).flatMap(parseDate)
-
-        if(start.isEmpty && end.isEmpty) None
-        else Some(DateRange(from = start, to = end))
+        Some(DateRange(
+          from = getOptionFromQS("startDate", qs).flatMap(parseDate).getOrElse(DateTime.now().minusYears(10)),
```

Comment:
Ok cool. Maybe pull `DateTime.now().minusYears(10)` out into a variable and call it something like `baseMetricsDate`? Or a better name than that